src/main/java/com/example/petbuddybackend/service/chat/ChatService.java

```diff
@@ -90,20 +94,25 @@ public ChatMessageDTO createMessage(
                 chatRoom.getCaretaker().getAccountData();
 
         ChatMessage persistedMessage = persistMessage(chatRoom, sender, chatMessage.getContent(), seenByRecipient);
-        performLastMessageSeenUpdate(chatRoom.getId(), senderEmail, senderRole, seenByRecipient);
+        performPreviousMessagesSeenUpdate(chatRoom.getId(), senderEmail, senderRole, seenByRecipient);
+
+        String receiverEmail = getMessageReceiverEmail(chatRoom, senderRole);
+        sendUnseenChatsNotification(receiverEmail);
+
```

Comment:
Nie wiem czy bym tej logiki na websockety nie przeniósł. Tam już leci update że user coś przeczytał. Tutaj to będzie lecieć powiadomienie za każdym razem jak użytkownik scrolluje po wiadomości. Plus websockety już updateują to czy odczytałeś wiadomość jak jesteś na chacie

Reply:
Jedynym wyjątkiem do updateu `seenByRecipient` to jest to wysyłanie wiadomości ale tutaj też musisz być podłączony do chatu żeby coś wysłać

Reply:
Dlatego wysylamy tylko liczbe nieodczytanych czatów. Jesli odbiorca jest w czacie to liczba sie nie zmieni i nic sie nie stanie

Reply:
A czy nie lepiej dać wywołanie tej funkcji na event subscribe do chatu?
`sendUnseenChatsNotification(receiverEmail);`

Bo właściwie wtedy faktycznie mogła by się zmienić ta liczba nieodczytanych wiadomości. A aktualnie to powiedzmy ktoś wejdzie sobie w chat, przescrolluje 1000 wiadomości i backend 100 razy będzie liczył wiadomości nieodczytane

Reply:
Znaczy ja to robię w metodzie gdzie wiadomosc wysyłamy. A rzeczywiście zapomniałem jeszcze dodać kiedy ktoś w czat pi prostu wejdzie żeby odświeżyć tą liczbę. Chyba że wtedy front sam sobie będzie zaciągał

Reply:
I być może rzeczywiście zrobiw to w eventach w controllerze. A i tak przy pierwszej wiadomości muszę w serwisie